src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobFindTree.java

```diff
@@ -104,12 +118,7 @@ public TreePathResult getResult()
     @Override
     protected double computeHeuristic(@NotNull final BlockPos pos)
     {
-        final int dx = pos.getX() - hutLocation.getX();
-        final int dy = pos.getY() - hutLocation.getY();
-        final int dz = pos.getZ() - hutLocation.getZ();
-
-        //  Manhattan Distance with a 1/1000th tie-breaker - halved
-        return (Math.abs(dx) + Math.abs(dy) + Math.abs(dz)) * TIE_BREAKER;
+        return furthestRestriction == null ? pos.distanceSq(hutLocation) * TIE_BREAKER : Math.sqrt(BlockPosUtil.getDistanceSquared2D(pos, furthestRestriction));
```

Comment:
not sure if sqrt is needed here, probably can be removed to save some computational time

Reply:
Not needed